src/Listeners/Yith.php

```diff
@@ -26,15 +30,139 @@ public function register_hooks() {
 	 * @return string The new option value
 	 */
 	public function paypal_connection( $new_option, $old_option ) {
-		if ( $new_option !== $old_option && ! empty( $new_option ) ) {
+		$url =  isset($_SERVER['HTTPS']) &&
+		$_SERVER['HTTPS'] === 'on' ? "https://" : "http://"; 
```

Comment:
We should use the `is_ssl()` in all places where we're currently doing this code. The WordPress function has more robust checks for SSL.

https://developer.wordpress.org/reference/functions/is_ssl/